sdks/python/apache_beam/internal/gcp/auth.py

```diff
@@ -159,12 +159,19 @@ def _get_service_credentials(pipeline_options):
           'Credentials.')
       return credentials
     except Exception as e:
-      _LOGGER.warning(
+      _LOGGER.error(
```

Comment:
I am not sure we need to change this to error since connecting anonymously is allowed. And can we also make the message much clearer? e.g, Unable to find default credentials to use: %s\n, Connecting anonymously. This is expected if no credentials are required to access GCP resources.